usm_rest_api/v1/views/utils.py

```diff
@@ -0,0 +1,234 @@
+HOST_TYPE_MIXED = 3
+HOST_TYPE_GLUSTER = 4
+HOST_STATUS_INACTIVE = 1
+HOST_STATUS_ACTIVE = 2
+
+
+def peer(hostlist, newNode):
+    rc = False
+    if hostlist:
+        host = random.choice(hostlist)
+
+        rc = salt_wrapper.peer(host, newNode)
+        if rc is True:
+            return rc
+        #
+        # random host is not able to peer probe
+        # Now try to iterate through the list of hosts
+        # in the cluster to peer probe until
+        # the peer probe is successful
+        #
+
+        # No need to send it to host which we already tried
+        hostlist.remove(host)
+        for host in hostlist:
+            rc = salt_wrapper.peer(host, newNode)
+            if rc is True:
+                return rc
+        return rc
+    else:
+        return True
+
+
+def create_cluster(cluster_data):
+    # create the cluster
+    cluster_data['cluster_id'] = str(uuid.uuid4())
+    try:
+        clusterSerilaizer = ClusterSerializer(data=cluster_data)
+        if clusterSerilaizer.is_valid():
+            clusterSerilaizer.save()
+        else:
+                log.error(
+                    "Cluster Creation failed. Invalid clusterSerilaizer")
+                log.error(
+                    "clusterSerilaizer Err: %s" % clusterSerilaizer.errors)
+                raise Exception(
+                    "Cluster Creation failed. Invalid clusterSerilaizer",
+                    clusterSerilaizer.errors)
+    except Exception, e:
+        log.exception(e)
+        raise e
+
+
+def setup_transport_and_update_db(cluster_data, nodelist):
+    log.debug("Inside setup_transport_and_update_db function")
+    failedNodes = []
+
+    # Create the Nodes in the cluster
+    # Setup the host communication channel
+    # current_task.update_state(state='ESTABLISH_HOST_COMMUNICATION')
+    minionIds = {}
+    for node in nodelist:
+        try:
+            ssh_fingerprint = usm_wrapper_utils.get_host_ssh_key(
+                node['management_ip'])
+            minionIds[node['management_ip']] = salt_wrapper.setup_minion(
+                node['management_ip'], ssh_fingerprint[0],
+                node['ssh_username'], node['ssh_password'])
+        except Exception, e:
+            log.exception(e)
+            failedNodes.append(node)
+    log.debug(minionIds)
+    # Sleep for sometime to make sure all the restarted minions are back
+    # online
+    time.sleep(3)
+    log.debug("Accepting the minions keys")
+    # current_task.update_state(state='ADD_MINION_KEYS_AND_DB_UPDATE')
+
+    # Accept the keys of the successful minions and add to the DB
+    for node in nodelist:
+        try:
+            if node not in failedNodes:
+                salt_wrapper.accept_minion(minionIds[node['management_ip']])
+        except Exception, e:
+            log.exception(e)
+            failedNodes.append(node)
+    log.debug(minionIds)
+    #
+    # Wait for some time so that the communication chanel is ready
+    #
+    time.sleep(15)
```

Comment:
This sleep got longer. Would you please turn this into a module level "constant" with a TODO comment on how you expect this to go away?